File: schedule-service/src/main/java/pe/albrugroup/schedule_service/entity/response/horario/HorarioResponse.java
package pe.albrugroup.schedule_service.entity.response.horario;

import lombok.*;
import pe.albrugroup.schedule_service.entity.enums.ModalidadContrato;

import java.time.LocalDate;
import java.util.List;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class HorarioResponse {

    private Long id;
    private Long idEmpleado;
    private Long idContrato;
    private ModalidadContrato modalidad;
    private Integer horasObjetivoSemanal;
    private Integer horasObjetivoMensual;
    private Integer minutosAlmuerzo;
    private Integer minutosServicios;
    private LocalDate fechaInicio;
    private LocalDate fechaFin;
    private Boolean compensable;
    private List<HorarioDetalleResponse> detalles;
    private List<ExcepcionHorarioResponse> excepciones;
}

File: frontend/src/app/shared/models/schedule/horario-response.ts
export interface HorarioResponse {
  id: number;
  idEmpleado: number;
  idContrato: number;
  modalidad: string;
  horasObjetivoSemanal: number;
  horasObjetivoMensual: number;
  minutosAlmuerzo: number;
  minutosServicios: number;
  fechaInicio: string;
  fechaFin: string | null;
  compensable: boolean;
  detalles: Array<{
    id: number;
    dia: string;
    horaEntrada: string;
    horaSalida: string;
    inicioAlmuerzo: string;
    finAlmuerzo: string;
    laborable: boolean;
  }>;
  excepciones: unknown[];
}
